serde-jackson/src/main/java/io/micronaut/serde/jackson/JacksonJsonMapper.java

```diff
@@ -128,8 +128,9 @@ private JacksonJsonMapper(@NonNull SerdeRegistry registry,
     @NonNull
     @Override
     public JsonMapper createSpecific(@NonNull Argument<?> type) {
+        JacksonJsonMapper mapper;
```

Comment:
It looks to me that having the view in the mapper is useless. 
I think it should be extracted in `ObjectDeserializer/ObjectSerializer` and create a new serializer /deserializer wrapper that will create a new context with something like `context = context.withViews(...)`

but that will not add a view for custom serdes so I don't know.

Reply:
i think it is sensible that the view is in the encoder context. i dont think a serializer wrapper is the right place to modify the context depending on view, the fact that we change the context view based on the argument is very specific to the JsonMappers so that is where it should stay. If we included this extractView logic inside a serializer, we risk using the JsonView of a property to modify the encoder context, which is not what a JsonView on a property is supposed to do (it is supposed to modify the visibility of the property, not change the view for child properties). And of course there is the technical issue that we'd need to change this for every createSpecific implementation unless we start wrapping literally every serde.